stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkPeerManager.java

```diff
@@ -50,9 +51,19 @@ public AsyncPrettyPrinterRegistry getAsyncPrettyPrinterRegistry() {
     return mAsyncPrettyPrinterRegistry;
   }
 
+  public void setPrettyPrinterInitializer(AsyncPrettyPrinterInitializer initializer) {
+    mPrettyPrinterInitializer = Util.throwIfNull(initializer);
+  }
+
   private final PeersRegisteredListener mTempFileCleanup = new PeersRegisteredListener() {
     @Override
     protected void onFirstPeerRegistered() {
+      if (mAsyncPrettyPrinterRegistry == null) {
```

Comment:
you should shortcut this. 
`if (mAsyncPrettyPrinterRegistry == null && mPrettyPrinterInitializer != null)`

There is no need to reinitialize if you have already done it. See my note in `#setPrettyPrinterInitializer()`